account/account_test.go

```diff
@@ -530,6 +532,51 @@ func TestTransactionHashDeclare(t *testing.T) {
 	require.Equal(t, expectedHash.String(), hash.String(), "TransactionHashDeclare not what expected")
 }
 
+func TestWaitForTransactionReceiptMOCK(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+	mockRpcProvider := mocks.NewMockRpcProvider(mockCtrl)
+
+	mockRpcProvider.EXPECT().ChainID(context.Background()).Return("SN_GOERLI", nil)
+	acnt, err := account.NewAccount(mockRpcProvider, &felt.Zero, "", starknetgo.NewMemKeystore())
+	require.NoError(t, err, "error returned from account.NewAccount()")
+
+	type testSetType struct {
+		Hash            *felt.Felt
+		ExpectedErr     error
+		ExpectedReceipt rpc.TransactionReceipt
+	}
+	testSet := map[string][]testSetType{
+		"mock": {
+			{
+				Hash:            new(felt.Felt).SetUint64(1),
+				ExpectedReceipt: nil,
+				ExpectedErr:     errors.New("UnExpectedErr"),
+			},
+			{
+				Hash: new(felt.Felt).SetUint64(2),
+				ExpectedReceipt: rpc.InvokeTransactionReceipt{
+					TransactionHash: new(felt.Felt).SetUint64(2),
+					ExecutionStatus: rpc.TxnExecutionStatusSUCCEEDED,
+				},
+				ExpectedErr: nil,
+			},
+		},
+	}[testEnv]
+
+	for _, test := range testSet {
+		mockRpcProvider.EXPECT().TransactionReceipt(context.Background(), test.Hash).Return(test.ExpectedReceipt, test.ExpectedErr)
+		resp, err := acnt.WaitForTransactionReceipt(context.Background(), test.Hash, 2*time.Second)
+
+		if err != nil {
+			require.Equal(t, test.ExpectedErr, err)
+		} else {
+			require.Equal(t, test.ExpectedReceipt.GetExecutionStatus(), (*resp).GetExecutionStatus())
```

Comment:
wdyt about making condition on expected behaviour rather than actual result:
```
if test.ExpectedErr != nil {
   ...
} else {
   ...
}
```
Pros is that if implementation for some reason will return nil error and it will be unexpected by your testdata (i.e. test.ExpectedReceipt will be nil) then line 574 will throw nil pointer dereference panic instead of asserting equality

Reply:
Good point!